webserver/__init__.py

```diff
@@ -20,24 +31,30 @@ def create_app_with_configuration(config_path=None):
     )
 
     # Configuration
-    root_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'default_config.py'
-    ))
-    app.config.from_pyfile(os.path.join(
-        root_path,
-        'custom_config.py'
-    ), silent=True)
+    config_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "config.py")
+    print(config_file)
+    if deploy_env and not config_path:
+        print("Checking if consul generated config file exists: $s" % config_file)
+        for _ in range(CONSUL_CONFIG_FILE_RETRY_COUNT):
+            if not os.path.exists(config_file):
+                time.sleep(1)
+
+        if not os.path.exists(config_file):
```

Comment:
as I understand, this loop is to wait for the config file to be created by consul, right? Out of interest, is there a reason why we're not using something like dockerize to wait for the file before launching flask? (https://github.com/jwilder/dockerize#waiting-for-other-dependencies)
The advantage of this is that we don't have this weird deploy-level logic inside a configuration/setup method

Reply:
Upon sleeping on it, I prefer to keep this as python code, complete with access for easy Sentry reporting in case shit goes wrong. Dockerize is good for a few things, namely ensuring the correct start-up order of containers. But I feel that this situation needs more nuance and more reporting.